lightning/src/ln/channelmanager.rs

```diff
@@ -7533,14 +7576,31 @@ where
 
 				// Now that we know we have a channel, assign an outbound SCID alias.
 				let outbound_scid_alias = self.create_and_insert_outbound_scid_alias();
-				channel.context.set_outbound_scid_alias(outbound_scid_alias);
+				channel_phase.context_mut().set_outbound_scid_alias(outbound_scid_alias);
 
-				peer_state.pending_msg_events.push(events::MessageSendEvent::SendAcceptChannel {
-					node_id: channel.context.get_counterparty_node_id(),
-					msg: channel.accept_inbound_channel(),
-				});
-
-				peer_state.channel_by_id.insert(temporary_channel_id.clone(), ChannelPhase::UnfundedInboundV1(channel));
+				match channel_phase {
+					ChannelPhase::UnfundedInboundV1(mut channel) => {
+						peer_state.pending_msg_events.push(events::MessageSendEvent::SendAcceptChannel {
+							node_id: channel.context.get_counterparty_node_id(),
+							msg: channel.accept_inbound_channel() });
+						peer_state.channel_by_id.insert(*temporary_channel_id,
+							ChannelPhase::UnfundedInboundV1(channel));
+					},
+					ChannelPhase::UnfundedInboundV2(mut channel) => {
+						peer_state.pending_msg_events.push(events::MessageSendEvent::SendAcceptChannelV2 {
+							node_id: channel.context.get_counterparty_node_id(),
+							msg: channel.accept_inbound_dual_funded_channel() });
+						peer_state.channel_by_id.insert(channel.context.channel_id(),
+							ChannelPhase::UnfundedInboundV2(channel));
+					},
+					_ => {
+						debug_assert!(false);
+						// This should be unreachable, but if it is then we would have dropped the inbound channel request
+						// and there'd be nothing to clean up as we haven't added anything to the channel_by_id map yet.
+						return Err(APIError::APIMisuseError {
+							err: "Channel somehow changed to a non-inbound channel before accepting".to_owned() })
+					}
```

Comment:
Probably can do a similar changes here as commented on in `internal_open_channel` to avoid this branch.